code/datums/character_preview.dm

```diff
@@ -0,0 +1,119 @@
+datum/character_preview
+	var/global/max_preview_id = 0
+	var/preview_id
+	var/window_id
+	var/client/viewer
+	var/obj/screen/handler
+	var/mob/living/carbon/human/preview_mob
+
+	window
+		New(client/viewer)
+			var/winid = "preview_[max_preview_id]"
+
+			winclone(viewer, "blank-map", winid)
+
+			winset(viewer, winid, list2params(list(
+				"size" = "128,128",
+				"title" = "Character Preview",
+				"can-close" = FALSE,
+				"can-resize" = FALSE,
+			)))
+
+			. = ..(viewer, winid)
+
+		disposing()
+			. = ..()
+			SPAWN_DBG(0)
```

Comment:
needed?

Reply:
It was in the viewport code, and I copied it from there.